base36.go

```diff
@@ -0,0 +1,158 @@
+/*
+
+Package base36 provides a reasonably fast implementation of a binary base36 codec.
+
+*/
+package base36
+
+// Simplified code based on https://godoc.org/github.com/mr-tron/base58
+// which in turn is based on https://github.com/trezor/trezor-crypto/commit/89a7d7797b806fac
+
+import (
+	"fmt"
+	"math"
+)
+
+const UcAlphabet = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+const LcAlphabet = "0123456789abcdefghijklmnopqrstuvwxyz"
+
+var encExpansion int
+var revAlphabet [256]uint64
+
+func init() {
+	for i := range revAlphabet {
+		revAlphabet[i] = 36
+	}
+	for i, c := range UcAlphabet {
+		revAlphabet[byte(c)] = uint64(i)
+		if c > '9' {
+			revAlphabet[byte(c)+32] = uint64(i)
+		}
+	}
+
+	encExpansion = int(
```

Comment:
This is going to be really in accurate. It's 2.

Reply:
We could use `127/82` as an over-approximation?

Reply:
Or we can just use 2.

Reply:
It's [what the original code used](https://github.com/mr-tron/base58/blob/master/base58.go#L36) so I replicated it. Doesn't hurt to overallocate.

Reply:
The original code estimates the size of the resulting buffer by first multiplying by 138, then dividing by 100, then adding 1. This just defines a constant multiple of 2.

Reply:
That's fine, but we might as well just define it as 2 if we're going to do that.

Reply:
Yeah I realized what you said after I implemented it, pushing up in a moment ;)